pandas/core/arrays/base.py

```diff
@@ -301,6 +319,60 @@ def unique(self):
         uniques = unique(self.astype(object))
         return self._constructor_from_sequence(uniques)
 
+    def _values_for_factorize(self):
+        """Return an array suitable for factorization.
+
+        Returns
+        -------
+        ndarray
+            An array suitable for factoraization. This should maintain order
+            and be a supported dtype.
+
+        """
+        return self.astype(object)
+
+    def factorize(self, na_sentinel=-1):
```

Comment:
you need a sort=False arg here

Reply:
Do we want / need that? It complicates the implementation a bit. Any idea what it's actually used for?